docs/dev/how-to/estimate-gas.md

```diff
@@ -0,0 +1,58 @@
+- [L2 to L2](#l2-to-l2) transactions.
+
+## L1 to L1
+
+To estimate gas for L1 to L1 transactions, use the Ethereum `eth_estimateGas` method and send to an L1 node.
+
+:::info More info
+For more information and live testing, check out the [Ethereum JSON RPC docs](https://ethereum.github.io/execution-apis/api-documentation/).
+:::
+
+## L1 to L2
+
+To estimate gas for an L1 to L2 transaction, first gather the required values:
+
+1. Get the current L1 gas price. 
+    
+    For example, the Ethers.js library has a [`getGasPrice()`](https://docs.ethers.org/v5/api/providers/provider/#Provider-getGasPrice) method available on a `Provider` object. Your language of choice should have an equivalent.
+
+2. Call the [`zks_estimateGasL1ToL2`](../../api/api.md#zks_estimategasl1tol2) method, passing the transaction data. This returns the estimated amount of gas the transaction requires; commonly called **gas limit** or similar in our code and docs.
+
+    - Apply an alias to the addresses in the request if the sender address is a contract. If the sender is an EOA, no aliasing is required. This is implemented by the [`applyL1ToL2Alias`](https://github.com/matter-labs/zksync-2-contracts/blob/7b5c094a57c0606785ea38c9c752f9def9a5ed9d/ethereum/contracts/vendor/AddressAliasHelper.sol#L28) Solidity function.
+
+3. Call the [`l2TransactionBaseCost`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/interfaces/IMailbox.sol#L129) function, passing the gas price and gas limit from previous steps. This function returns the base cost required to send a transaction.
+
+    - You also need to pass the constant representing how much gas is required to publish a byte of data from L1 to L2. At the time of writing, the JavaScript API provides this constant as [`REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT`](../../api/js/utils.md#gas).
+
+Once you have all of the above, you can send a transaction with the [`requestL2Transaction`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/interfaces/IMailbox.sol#L119) function, passing the gas limit returned at step 2, along with the gas price and base cost from steps 1 and 3 as part of the value parameters.
+
+For a more detailed explanation on estimating gas, with code examples, find out [how to send a transaction from L1 to L2](../how-to/send-transaction-l1-l2.md).
+
+
+## L2 to L2
+
+Gas estimation for L2 to L2 transactions on zkSyncEra works in the same way as it does in Ethereum.
+
+Supply the same call data as an L1 to L2 transaction, use the Ethereum `eth_estimateGas` method, and send the transaction to an L2 node.
+
+:::info More info
+For more information and live testing, check out the [Ethereum JSON RPC docs](https://ethereum.github.io/execution-apis/api-documentation/).
+:::
+
+## L2 to L1
+
+Gas estimation is dealt with implicitly by the SDK for L2 to L1 transactions.
```

Comment:
Better to say that all the L2->L1 messages are parts of the corresponding L2->L2 transactions

Reply:
Remove mention of gas estimation?
Or something like **gas estimation for L2->L1 messages is dealt with by the corresponding L2->L2 txs** ??

Reply:
@barakshani ?

Reply:
I'm not the one who made the initial comment.

Reply:
All input is welcome to get this out the door.

Reply:
@StanislavBreadless are you happy with the bolded sentence above?

Reply:
@kailash-manasarovar yes